cloudpathlib/cloudpath.py

```diff
@@ -245,10 +245,8 @@ def __del__(self) -> None:
             self._handle.close()
 
         # ensure file removed from cache when cloudpath object deleted
-        if (
-            hasattr(self, "client")
-            and self.client.file_cache_mode == FileCacheMode.cloudpath_object
-        ):
+        client = getattr(self, "client", object)
+        if getattr(client, "file_cache_mode", None) == FileCacheMode.cloudpath_object:
```

Comment:
(Not something I feel strongly about, mostly just a comment/nit.)

I haven't seen this pattern of using the `object` class as a sentinel/dummy value before. Is this common? I would've expected `None` instead.

Personally, I would find something closer to the original structure most readable, like:

```python
        if (
            hasattr(self, "client")
            and getattr(self.client, "file_cache_mode", None) == FileCacheMode.cloudpath_object
        ):
```

Reply:
It's a really good question 😄 My thought was f the `client` was `None` then you'd have to have the `and` condition. But now that I think about it `getattr(None, "file_cache_mode", None)` is the same as `getattr(object, "file_cache_mode", None)` so I think `object` is not needed. However, I did want to avoid the too-long-line (or another awkward break in the condition) with the original structure, which is why I created the intermediate variable `client`. Maybe something like:
```python
file_cache_mode = getattr(getattr(self, "client", None), "file_cache_mode", None)
if file_cache_mode == FileCacheMode.cloudpath_object:
```
?

Reply:
I just pushed a commit that changes `object` to `None`. This is probably not worth bikeshedding any more about.